compass/landice/mesh.py

```diff
@@ -753,3 +764,344 @@ def make_region_masks(self, mesh_filename, mask_filename, cores, tags):
+    data.variables['x'][:] = x1
+    data.variables['y'][:] = y1
+
+    data.close()
+
+    return preprocessed_gridded_dataset
+
+
+def interp_ais_bedmachine(self, data_path, mali_scrip, nProcs, dest_file):
+    """
+    Interpolates BedMachine thickness and bedTopography dataset
+    to a MALI mesh
+
+    Parameters
+    ----------
+    data_path : str
+        path to AIS datasets, including BedMachine
+
+    mali_scrip : str
+        name of scrip file corresponding to destination MALI mesh
+
+    nProcs : int
+        number of processors to use for generating remapping weights
+
+    dest_file: str
+        MALI input file to which data should be remapped
+    """
+
+    logger = self.logger
+
+    logger.info('creating scrip file for BedMachine dataset')
+    args = ['create_SCRIP_file_from_planar_rectangular_grid.py',
+            '-i',
+            os.path.join(data_path,
+                         'BedMachineAntarctica_2020-07-15_v02_edits_floodFill_extrap_fillVostok.nc'),  # noqa
+            '-s',
+            os.path.join(data_path,
+                         'BedMachineAntarctica_2020-07-15_v02.scrip.nc'),
+            '-p', 'ais-bedmap2',
+            '-r', '2']
+    check_call(args, logger=logger)
+
+    # Generate remapping weights
+    # Testing shows 5 badger/grizzly nodes works well.
+    # 2 nodes is too few. I have not tested anything in between.
+    logger.info('generating gridded dataset -> MPAS weights')
+    args = ['srun', '-n', nProcs, 'ESMF_RegridWeightGen',
+            '--source',
+            os.path.join(data_path,
+                         'BedMachineAntarctica_2020-07-15_v02.scrip.nc'),
+            '--destination', mali_scrip,
+            '--weight', 'BedMachine_to_MPAS_weights.nc',
+            '--method', 'conserve',
+            "--netcdf4",
+            "--dst_regional", "--src_regional", '--ignore_unmapped']
+    check_call(args, logger=logger)
+
+    # Perform actual interpolation using the weights
+    logger.info('calling interpolate_to_mpasli_grid.py')
+    args = ['interpolate_to_mpasli_grid.py', '-s',
+            os.path.join(data_path,
+                         'BedMachineAntarctica_2020-07-15_v02_edits_floodFill_extrap_fillVostok.nc'),  # noqa
+            '-d', dest_file,
+            '-m', 'e',
+            '-w', 'BedMachine_to_MPAS_weights.nc']
+    check_call(args, logger=logger)
+
+
+def interp_ais_measures(self, data_path, mali_scrip, nProcs, dest_file):
+    """
+    Interpolates MEASURES ice velocity dataset
+    to a MALI mesh
+
+    Parameters
+    ----------
+    data_path : str
+        path to AIS datasets, including BedMachine
+
+    mali_scrip : str
+        name of scrip file corresponding to destination MALI mesh
+
+    nProcs : int
+        number of processors to use for generating remapping weights
+
+    dest_file: str
+        MALI input file to which data should be remapped
+    """
+
+    logger = self.logger
+
+    logger.info('creating scrip file for velocity dataset')
```

Comment:
As with the BedMachine scrip file, let's check for an existing file in the data directory before writing a huge new one. And it should write any new file to the work directory, leaving the data directory untouched. If the scrip file exists in the data directory, let's just put a symlink to it in the work directory.

Reply:
I've modified to write the scrip file to the workdir.